languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/pt-BR/style.xml

```diff
@@ -952,6 +955,154 @@ USA
+                <suggestion>se <match no="2" postag_regexp="yes" postag="(V...)2S(.)" postag_replace="$13S$2"/></suggestion>
+                <example correction="se esqueça">Não <marker>te esqueças</marker>!</example>
+                <example correction="se esqueceu">Acho que <marker>te esqueceste</marker>...</example>
+            </rule>
+
+            <rule> <!-- reflexive enclitic -->
+                <pattern>
+                    <token postag_regexp="yes" postag="V...2S."/>
+                    <token regexp="yes" spacebefore="no">&hifen;</token>
+                    <token spacebefore="no">te</token>
+                </pattern>
+                <message>No português brasileiro, o uso da segunda pessoa do singular não é consistente em todos os dialetos. Seu texto soará mais neutro se substituir por &quot;te&quot;.</message>
+                <suggestion><match no="1" postag_regexp="yes" postag="(V...)2S(.)" postag_replace="$13S$2"/>-se</suggestion>
+                <example correction="Lembre-se"><marker>Lembra-te</marker>.</example>
+            </rule>
+
+            <rule> <!-- accusative proclitic -->
+                <pattern>
+                    <token>te</token>
+                    <token postag_regexp="yes" postag="V...[13]..">
+                        <exception regexp="yes" inflected="yes">&dative_person_verbs;</exception>
+                    </token>
+                </pattern>
+                <message>No português brasileiro, o uso da segunda pessoa do singular não é consistente em todos os dialetos. Seu texto soará mais neutro se substituir por &quot;você&quot;.</message>
+                <suggestion>\2 você</suggestion>
+                <example correction="vi você">Nunca <marker>te vi</marker> por lá.</example>
+                <example correction="pomos você">Sempre <marker>te pomos</marker> em primeiro lugar.</example>
+                <example>Já te dei um presente.</example>
+            </rule>
+
+            <rule> <!-- accusative enclitic; I'm not adding the 'a'/'o' options here because of the annoying transformations -->
+                <pattern>
+                    <token postag_regexp="yes" postag="V...[13]..">
+                        <exception regexp="yes" inflected="yes">&dative_person_verbs;</exception>
+                    </token>
+                    <marker>
+                        <token regexp="yes" spacebefore="no">&hifen;</token>
+                        <token spacebefore="no">te</token>
+                    </marker>
+                </pattern>
+                <message>No português brasileiro, o uso da segunda pessoa do singular não é consistente em todos os dialetos. Seu texto soará mais neutro se substituir por &quot;você&quot;.</message>
+                <suggestion> você</suggestion>
+                <example correction=" você">Conheço<marker>-te</marker> desde 2002.</example>
```

Comment:
I think "lhe" would be a better correction for this. It's not grammatically ideal, but neither is the original sentence for this example. Check diff results [here](https://regression.languagetoolplus.com/via-http/2023-06-30/pt-BR/result_style_INFORMAL_2SG_BR%5B10%5D.html)

"Conheço você" as a replacement for "te conheço", on the other hand, works better (and I see there is a subrule for this already)

Reply:
idgi... `lhe` there is not 'not ideal', it's prescriptively incorrect. My original idea was to have `-o`/`-a` there, since that'd be the 3SG equivalent, but there are issues with some verb endings (`-n[ao]`, `-l[ao]`).

afaics the bad diff results come from ditransitive verbs, so perhaps it might be enough to expand that list (I created an entity for that). Or even just having a separate rule for ditransitive verbs? Either way, I don't think it's a good idea to introduce a proscribed structure.